charts/prometheus-postgresql-alerts/values.yaml

```diff
@@ -11,7 +11,7 @@ rulesGroupName: postgresql.rules
 rules:
 
   SQLExporterDown:
-    expr: absent(up{service="sql-exporter", target=""})
+    expr: absent(up{service="sql-exporter", target=""}) or up{service="sql-exporter", target=""} <1
```

Comment:
I'm surprised by the `target=""` filters, won't this report that the metric is never there?

Reply:
> did you find out about this following a real use case where the exporter was down but still reporting the up metric?

Yes, there was such behaviour this morning. The pod was unstable, the exporter was CPU throttling and couldn't work as expected. 

This can be detected by the self export of up metric, which the sql-exporter is smart enough to set to 0 if it didn't work. This up metric is applied to no target, that's why we need to filter to target="". If the sql-exporter is up and running, this metric is also sent and set to 1. 

See for yourself in our Grafana dashboard named "SQL Exporter health" by setting a time range to 12h.